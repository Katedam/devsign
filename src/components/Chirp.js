import React from 'react';
import PropTypes from 'prop-types';
import { ProfileImg, Handle, Text, Article } from '../styles/mainStyle';
import styled from 'styled-components';

const HTwo = styled.h2 `
  margin: 0;
`;

const P = styled(Text) `
  margin: 0;
  padding: 0;
  padding-left: 13%;
`;

function Chirp({ name, handle, text, profileImage }) {
  return (
    <Article>
      <header>
        <ProfileImg alt={name} src={profileImage}/>
        <HTwo>{name} <Handle>{handle}</Handle></HTwo>
      </header>
      <P>{text}</P>
    </Article>
  );
}

Chirp.propTypes = {
  name: PropTypes.string.isRequired,
  handle: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  profileImage: PropTypes.string.isRequired
};

export default Chirp;
