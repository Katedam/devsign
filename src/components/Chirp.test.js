import React from 'react';
import Chirp from './Chirp';
import renderer from 'react-test-renderer';

describe('Chirp component', () => {
  it('matches a snapshot', () => {
    const chirp = {
      profileImage: 'http://cdn.onlinewebfonts.com/svg/img_191958.png',
      name: 'Kaiya',
      handle: '@ladybeard',
      text: 'chirp chirp chirp'
    };
    const tree = renderer.create(
      <Chirp
        profileImage={chirp.profileImage}
        name={chirp.name}
        handle={chirp.handle}
        text={chirp.text}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
