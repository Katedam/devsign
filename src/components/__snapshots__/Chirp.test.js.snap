// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chirp component matches a snapshot 1`] = `
<article
  className="sc-jzJRlG dFsfHb"
>
  <header
    className="sc-chPdSV iJfIXb"
  >
    <img
      alt="Kaiya"
      className="sc-jTzLTM fkBueS"
      src="http://cdn.onlinewebfonts.com/svg/img_191958.png"
    />
    <h2
      className="sc-cSHVUG eRmrdv"
    >
      Kaiya
       
      <span
        className="sc-fjdhpX FzRiJ"
      >
        @ladybeard
      </span>
    </h2>
  </header>
  <p
    className="sc-gqjmRU sc-kAzzGY ctCWvU"
  >
    chirp chirp chirp
  </p>
</article>
`;
